src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -267,28 +336,57 @@ newtype MaxInFlight = MaxInFlight Int
 -- not the same of the message we're sending.
 type DequeuePolicy = NodeType -> Dequeue
 
-defaultDequeuePolicy :: NodeType -- ^ Our node type
-                     -> DequeuePolicy
-defaultDequeuePolicy NodeCore = go
+-- | Pick default dequeue policy for a given node
+--
+-- NOTE: Assumes standard behind-NAT in the case of edge nodes.
+defaultDequeuePolicy :: NodeType -> DequeuePolicy
+defaultDequeuePolicy NodeCore  = defaultDequeuePolicyCore
+defaultDequeuePolicy NodeRelay = defaultDequeuePolicyRelay
+defaultDequeuePolicy NodeEdge  = defaultDequeuePolicyEdgeBehindNat
+
+-- | Default dequeue policy for core nodes
+defaultDequeuePolicyCore :: DequeuePolicy
+defaultDequeuePolicyCore = go
   where
-    -- Dequeueing policy for core nodes
     go :: DequeuePolicy
-    go NodeCore  = Dequeue NoRateLimiting (MaxInFlight 2)
-    go NodeRelay = Dequeue NoRateLimiting (MaxInFlight 1)
+    go NodeCore  = Dequeue NoRateLimiting (MaxInFlight 3)
+    go NodeRelay = Dequeue NoRateLimiting (MaxInFlight 2)
```

Comment:
I'm happy to have this change for now, until we can resolve and test the re-entrant conversation business.